### DISCUSSION TITLE
Is it possible to include an IP address in URL path?

### DISCUSSION THREAD

**Discussion by hussam-qasem** [mojolicious/mojo#2103] [Q&A] [answered]
I asked this [question on stack overflow](https://stackoverflow.com/questions/77076369/openapi-is-it-possible-to-include-ip-address-string-with-dots-in-url-path) but gotten no responses yet, so I apologize for the repetition.

Using [`Mojolicious::Plugin::OpenAPI`](https://metacpan.org/pod/Mojolicious::Plugin::OpenAPI), I'm trying to include an IP address in the URL path (not query parameter) but when I try `GET /api/echo/192.168.1.1`, I get an http/404 `Not Found` error . Here's my full app:

`./api.yaml`
```yaml
---
openapi: 3.0.3
servers:
  - url: /api
info:
  version: 1
  title: MyAPI
paths:
  /echo/{ip}:
    get:
      x-mojo-to: Example#echo
      parameters:
      - name: ip
        in: path
        required: true
        schema:
          type: string
          format: ipv4
      responses:
        '200':
          description: OK
        '500':
          description: Internal Server Error
```

`./lib/MyApp.pm`
```perl
package MyApp;
use Mojo::Base 'Mojolicious', -signatures;

sub startup ($self) {
  $self->plugin(OpenAPI => {url=>$self->home->rel_file("api.yaml")});
  $self->plugin(SwaggerUI => {route=>$self->routes()->any('/swagger'),url=>"/api"});
}

1;
```

`./lib/MyApp/Controller/Example.pm`
```perl
package MyApp::Controller::Example;
use Mojo::Base 'Mojolicious::Controller', -signatures;

sub echo($self) {
  my $app   = $self->openapi->valid_input or return;
  my $input = $app->validation->output;
  my $ip    = $input->{ip};
  $app->render(openapi=>{ip=>$ip});
}

1;
```

**Comment by rabbiveesh** [comment, marked as answer]
I imagine the issue is that the route being created is using a [standard placeholder](https://metacpan.org/dist/Mojolicious/view/lib/Mojolicious/Guides/Routing.pod#Standard-placeholders) which don't parse anything after the dot.
From looking at [the docs for the plugin](https://metacpan.org/dist/Mojolicious-Plugin-OpenAPI/view/lib/Mojolicious/Plugin/OpenAPI/Guides/OpenAPIv3.pod#Specification), it seems like you want to use the `x-mojo-placeholder` option to define which placeholder type you need

**Reply by kraih** [reply]
The [routes](https://docs.mojolicious.org/Mojolicious/Guides/Routing#Introspection) command should also show what kind of route the plugin generates.
